app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -5,88 +5,168 @@ import androidx.compose.runtime.mutableStateOf
+                                .filterIsInstance<ObserveConversationDetailsUseCase.Result.Success>() // TODO handle StorageFailure
+                                .map { it.conversationDetails }
+                            combine(
+                                conversationDetailsFlow,
+                                isSelfAdminFlow,
+                                getSelfTeam()
+                            ) { conversationDetails, isSelfAnAdmin, selfTeam ->
+                                Triple(
+                                    conversationDetails,
+                                    isSelfAnAdmin,
+                                    selfTeam
                                 )
+                            }.collect { (conversationDetails, isSelfAnAdmin, selfTeam) ->
+                                with(conversationDetails) {
+                                    val isSelfInOwnerTeam =
+                                        selfTeam?.id != null && selfTeam.id == conversationDetails.conversation.teamId?.value
+
+                                    if (this is ConversationDetails.Group) {
+                                        updateState(
+                                            groupOptionsState.copy(
+                                                groupName = conversation.name.orEmpty(),
+                                                isUpdatingAllowed = isSelfAnAdmin,
+                                                areAccessOptionsAvailable = conversation.isTeamGroup(),
+                                                isGuestAllowed = (conversation.isGuestAllowed() || conversation.isNonTeamMemberAllowed()),
+                                                isServicesAllowed = conversation.isServicesAllowed(),
+                                                isUpdatingGuestAllowed = isSelfAnAdmin && isSelfInOwnerTeam,
+                                                isAbleToRemoveGroup = selfUser.teamId != null
+                                                        && conversation.creatorId.value == selfUser.id.value
+                                            )
+                                        )
+                                    }
+                                }
                             }
                         }
-                    }
                 }
         }
     }
 
+    fun leaveGroup() {
+        viewModelScope.launch {
+            requestInProgress = true
+            val response = withContext(dispatcher.io()) {
+                val selfUser = observerSelfUser().first()
+                removeMemberFromConversation(
+                    conversationId,
+                    selfUser.id
+                )
+            }
+            when (response) {
+                is RemoveMemberFromConversationUseCase.Result.Failure ->
+                    showSnackBarMessage(response.cause.uiText())
```

Comment:
Is this the way we really want to handle errors? we wouldn't be able to localize them like this. If this is not final solution, then maybe should we add a TODO ?